### PR TITLE
Enhancement: Streamline Docs Building

### PR DESCRIPTION
**Pull request by brolewis**
Auto generate the Sphinx `conf.py`, allowing for one source of truth (`pyproject.toml`) and move `index.rst` to the root folder as `jgt_tools.index` so that it doesn't have to be in it's own folder.

### CODE REVIEW

File: .jgt_tools.index [hunk 1]
```diff
@@ -14,12 +14,12 @@ Default values for each are:
...
-==================================
+========================
 
 .. _api-documentation:
 
 JGT Tools - How To Use It - Module Documentation
-----------------------------------------------------------
+------------------------------------------------
 
 For users of this package, the nitty gritty coding and configuration details.
```
Comment (brolewis) [inline .jgt_tools.index]: @bradsbrown and I discussed making the custom bits stored elsewhere (`pyproject.toml`?) and optional and build the `index.rst` dynamically, but felt that would be a FF to this PR.

Comment (dgou) [reply .jgt_tools.index]: Yeah, in some product specific repos there is a dynamicity to how parts of that file are created.
It's not immediately obvious to me that a `pyproject.toml` file is a good place for embedding `.rst` content...

**Comment by bradsbrown** [comment]
With 2 approvals, merging

File: .jgt_tools.index (hunk 1 above)
Comment (brolewis) [reply .jgt_tools.index]: There's a reason I ended that parenthetical with a question mark. The bigger improvement, to me, is to not *require* a user to have an index file; only leverage the ability to customize it, if so desired.

**Pull request merged**